components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/AuthenticatorManagementDAO.java

```diff
@@ -0,0 +1,93 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.dao;
+
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtException;
+import org.wso2.carbon.identity.application.common.model.LocalAuthenticatorConfig;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.AuthenticationType;
+
+import java.util.List;
+
+/**
+ * This interface performs CRUD operations for the Local Application Authenticator configurations.
+ */
+public interface AuthenticatorManagementDAO {
+
+    /**
+     * Create a new user defined Local Application Authenticator configuration.
+     *
+     * @param authenticatorConfig   Local Application Authenticator configuration.
+     * @param tenantId              Tenant Id.
+     *
+     * @return Created LocalAuthenticatorConfig.
+     * @throws AuthenticatorMgtException If an error occurs while adding the authenticator configuration.
+     */
+    LocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
+            LocalAuthenticatorConfig authenticatorConfig, Integer tenantId, AuthenticationType type)
+            throws AuthenticatorMgtException;
+
+    /**
+     * Update a user defined Local Application Authenticator configuration.
+     *
+     * @param existingAuthenticatorConfig   Existing  Local Application Authenticator configuration.
+     * @param updatedAuthenticatorConfig    New Local Application Authenticator configuration.
+     * @param tenantId                      Tenant Id.
+     *
+     * @return Updated LocalAuthenticatorConfig.
+     * @throws AuthenticatorMgtException If an error occurs while updating the authenticator configuration.
+     */
+    LocalAuthenticatorConfig updateUserDefinedLocalAuthenticator(LocalAuthenticatorConfig existingAuthenticatorConfig,
+            LocalAuthenticatorConfig updatedAuthenticatorConfig, Integer tenantId)
+            throws AuthenticatorMgtException;
+
+    /**
+     * Retrieve a Local user defined Application Authenticator configuration by name.
+     *
+     * @param authenticatorConfigName   Name of the Local Application Authenticator configuration.
+     * @param tenantId                  Tenant Id.
+     *
+     * @return Retrieved LocalAuthenticatorConfig
+     * @throws AuthenticatorMgtException If an error occurs while retrieving the authenticator configuration.
+     */
+    LocalAuthenticatorConfig getUserDefinedLocalAuthenticator(String authenticatorConfigName, Integer tenantId)
+            throws AuthenticatorMgtException;
+
+
+    /**
+     * Retrieve all user defined Local Application Authenticator configurations.
+     *
+     * @param tenantId                  Tenant Id.
+     *
+     * @return Retrieved LocalAuthenticatorConfig
+     * @throws AuthenticatorMgtException If an error occurs while retrieving the authenticator configurations.
+     */
+    List<LocalAuthenticatorConfig> getAllUserDefinedLocalAuthenticator(Integer tenantId)
```

Comment:
Shouldn't this be "getAllUserDefinedLocalAuthenticators" ?

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8